src/lib/network/network_autoyast.rb

```diff
@@ -106,24 +108,63 @@ def set_network_service
     #
-    # FIXME: it currently doesn't write DNS configuration. It is used for initialization
-    # of DNS setup according AY profile in 1st stage as part of network setup was moved
-    # here already and some parts of network configuration needs to know it. DNS write
-    # is still done in 2nd stage.
+    # If the installer is running in 1st stage mode only, then the configuration
+    # is also written
     #
+    # @param [Boolean] write forces instant writing of the configuration
     # @return [Boolean] true when configuration was present and loaded from the profile
-    def configure_dns
-      ay_dns_config = ay_networking_section["dns"]
+    def configure_dns(write: false)
+      log.info("NetworkAutoYast: DNS / Hostname configuration")
 
-      return false if !ay_dns_config
+      configure_submodule(DNS, ay_networking_section["dns"], write: write)
+    end
 
-      DNS.Import(ay_dns_config)
+    # Initializates Routing setup according AY profile
+    #
+    # If the installer is running in 1st stage mode only, then the configuration
+    # is also written
+    #
+    # @param [Boolean] write forces instant writing of the configuration
+    # @return [Boolean] true when configuration was present and loaded from the profile
+    def configure_routing(write: false)
+      log.info("NetworkAutoYast: Routing configuration")
 
-      log.info("NetworkAutoYast: DNS / Hostname configuration")
-      log.info("dhcp hostname: #{DNS.dhcp_hostname}")
-      log.info("write hostname: #{DNS.write_hostname}")
+      configure_submodule(Routing, ay_networking_section["routing"], write: write)
+    end
 
-      true
+    # Initializates NICs setup according AY profile
+    #
+    # If the installer is running in 1st stage mode only, then the configuration
+    # is also written
+    #
+    # @param [Boolean] write forces instant writing of the configuration
+    # @return [Boolean] true when configuration was present and loaded from the profile
+    def configure_nics(write: false)
+      log.info("NetworkAutoYast: NICs configuration")
+
+      ay_configuration = Lan.FromAY(ay_networking_section)
+
+      configure_submodule(LanItems, ay_configuration, write: write)
+    end
+
+    # Initializates /etc/hosts according AY profile
+    #
+    # If the installer is running in 1st stage mode only, then the configuration
+    # is also written
+    #
+    # @param [Boolean] write forces instant writing of the configuration
+    # @return [Boolean] true when configuration was present and loaded from the profile
+    def configure_hosts(write: false)
+      log.info("NetworkAutoYast: Hosts configuration")
+
+      # expected format for Hosts.Import is { "ip" => [list, of, names] }
+      hosts_config = (ay_host_section["hosts"] || {}).map do |host|
+        # we neet to guarantee order of the items here
+        [host["host_address"] || "", host["names"] || []]
+      end
+      hosts_config = hosts_config.to_h.delete_if { |k, v| k.empty? || v.empty? }
```

Comment:
This non-trivial FromAY-like section should go into its own method (class?) and be covered with tests.